core/domain/state_domain.py

```diff
@@ -3034,6 +3034,61 @@ def get_content_id_mapping_needing_translations(self, language_code):
 
         return content_id_to_html
 
+    def get_translated_text_and_ids(self, language_code):
+        """returns all the html that has been translated in a
+        a given language and their ids corresponding to translations
+
+        Args:
+            language_code: str. The abbreviated code of the language.
+
+        Returns:
+            list(tuple(str, str)). A list of tuples with first element
+            as content id and second element as the translation content html.
+        """
+        translated_content_ids = (
+            self.get_content_ids_that_are_correctly_translated(language_code))
+        translations_and_ids = []
+        for content_id in translated_content_ids:
+            translations_and_ids.append(
+                self.get_translated_content_with_content_id(
+                    content_id, language_code))
+
+        return translations_and_ids
+
+    def get_content_ids_that_are_correctly_translated(self, language_code):
```

Comment:
Creating a separate function for this doesn't seem to add much value here if the alternative is to just call `self.written_translations.get_content_ids_that_are_correctly_translated(language_code)` instead.